### DISCUSSION TITLE
Restoring chunk files from failed drive to two drives

### DISCUSSION THREAD

**Discussion by coderfi** [moosefs/moosefs#433] [Q&A]
This is similar to [Moving disks with chunks between chunkservers](https://github.com/moosefs/moosefs/discussions/425 )

One of my disks failed on a chunkserver and mfscgi reports several NO COPY chunks.

I have been able to recover the disk and copied some of the chunk files to a smaller disk, and the remainder to another disk (I didn't have a single large enough disk to take all of the original files).

Is it safe to simply add these two disks to the chunkserver (or any chunkserver) and add their paths to the mfshdd.cfg file (then reload the chunkserver).

How does moose know to which files these chunks belong?

I noticed there is a file called .metaid. Would I need to duplicate this across both disks?

thanks!
Fi

**Comment by jkiebzak** [comment]
Should be safe, as you described. No, do not duplicate .metaid

**Reply by coderfi** [reply]
Great!

Should I delete .metaid from both disks before reloading the chunkserver (and the chunkserver will create one appropriately)?

**Reply by oszafraniec** [reply]
you don't have to, chunkserver will detect changes as you added files manually and will perform full drive scan. Yes, you can move files by hand from out drive to another, you can move drives between chunkservers. Just dont do it with working chunkserver. Stop it, do what you have to, start chunkserver and it will do it's magic to see what's changed.

Look here for some info I wrote some time ago: https://github.com/moosefs/moosefs/issues/39

**Comment by chogata** [comment]
If you connect a disk with manually recovered chunks (from some other disk or disks) to your MooseFS, this disk should contain ONLY directories named from 00 to FF and the chunk files should be more or less evenly distributed into the directories. Do NOT copy anything else, no files starting with . (dot). Remember to set ownership and permissions to the directories and files correctly. After you add the path to the new disk to your mfshdd.cfg file, it's enough to reload the chunkserver process and it should "see" the drive and start scanning it to read the chunks.

**Comment by coderfi** [comment]
Thanks for the detailed notes! I’ll try this out, wish me luck!

On Tue, Apr 27, 2021 at 7:21 AM Agata Kruszona-Zawadzka <
***@***.***> wrote:

> If you connect a disk with manually recovered chunks (from some other disk
> or disks) to your MooseFS, this disk should contain ONLY directories named
> from 00 to FF and the chunk files should be more or less evenly distributed
> into the directories. Do NOT copy anything else, no files starting with .
> (dot). Remember to set ownership and permissions to the directories and
> files correctly. After you add the path to the new disk to your mfshdd.cfg
> file, it's enough to reload the chunkserver process and it should "see" the
> drive and start scanning it to read the chunks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/moosefs/moosefs/discussions/433#discussioncomment-664592>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDPY73SNA3CA3546TDPXYTTK3B5VANCNFSM43T5LPDQ>
> .
>
-- 
Fairiz "Fi" Azizi
